### PR TITLE
Add Player Block Place to the database

### PR DESCRIPTION
**Pull request by OverlordsIII**
Adds player block place events to the database and logged. Also refactors the LoggingEvent enum with a new type to specify that the player's pos was written instead of the block's pos in special cases.

### CODE REVIEW

File: src/main/java/tech/dttp/block/logger/BlockLogger.java [hunk 1]
```diff
@@ -10,22 +10,24 @@
 import tech.dttp.block.logger.util.LoggedEventType;
 
 public class BlockLogger implements ModInitializer {
+    public static DbConn db;
     @Override
     public void onInitialize() {
-        DbConn db = new DbConn();
```
Comment (yitzy299) [inline src/main/java/tech/dttp/block/logger/BlockLogger.java:15]: That's fine, just moving it

File: src/main/java/tech/dttp/block/logger/mixin/MixinServerPlayerInteractionManager.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+     *    - Check if what the player is looking at is a block
+     *    - if it is a block, then cast to a {@link BlockHitResult}
+     *    - Then write to the database the pos of the block result, the state we gathered earlier,
+     *      the player, the world the player used an item in, and the {@link LoggedEventType#PLACED}
+     *
+     *      In the rare condition of the player looking at the top part of the falling block, the player position
+     *      is logged instead of the block pos with {@link LoggedEventType#PLACED_PLAYER_POS}
+     *
+     *      The 2nd mixin just ensures that the interaction is written for creative and survival players
+     */
```
Comment (yitzy299) [inline src/main/java/tech/dttp/block/logger/mixin/MixinServerPlayerInteractionManager.java:41]: I feel that there isn't a need to log the player position, that induces more bugs that it solves. Could you change that?

File: src/main/java/tech/dttp/block/logger/util/LoggedEventType.java [hunk 3]
```diff
@@ -1,6 +1,7 @@
 package tech.dttp.block.logger.util;
 
 public enum LoggedEventType {
-    broken,
-    placed
+    BROKEN,
+    PLACED,
+    PLACED_PLAYER_POS
```
Comment (yitzy299) [inline src/main/java/tech/dttp/block/logger/util/LoggedEventType.java:6]: These here were lower case due to another bug involving Strings being buggy. Could those be changed back for compatibility reasons?

**Review by yitzy299 — CHANGES_REQUESTED** [review]
Generally good, please implement the changes I've suggested and I'll then merge

**Comment by yitzy299** [comment]
I also forgot to add: You didn't add the Mixin to blocklogger.mixins.json

File: src/main/java/tech/dttp/block/logger/mixin/MixinServerPlayerInteractionManager.java (hunk 2 above)
Comment (OverlordsIII) [reply src/main/java/tech/dttp/block/logger/mixin/MixinServerPlayerInteractionManager.java:41]: Ok im fixing it right now. Sorry about that

File: src/main/java/tech/dttp/block/logger/util/LoggedEventType.java (hunk 3 above)
Comment (OverlordsIII) [reply src/main/java/tech/dttp/block/logger/util/LoggedEventType.java:6]: Ok will change to lower case

**Comment by OverlordsIII** [comment]
Ok it should be fixed could you take a look again?

**Comment by yitzy299** [comment]
Also where is the event? I can't find an event interface file and I can't see any block places logged to the database

**Comment by OverlordsIII** [comment]
It doesn't create an actual event, it just adds the placement via a mixin instead of registering through an event.
it is wierd though that it isn't logging for you let me try again

**Comment by OverlordsIII** [comment]
this is what the placements look like for me

**Comment by yitzy299** [comment]
I can't get it working, check if you have uncommited changes or give me a video of it working

**Comment by OverlordsIII** [comment]
[here's the video](https://streamable.com/6iodks)

**Comment by yitzy299** [comment]
Turns out it's me
